From 2a3118ce4a9dc144237ebbfc0147fc535cd23c32 Mon Sep 17 00:00:00 2001
From: James Henstridge <james@jamesh.id.au>
Date: Mon, 2 Mar 2020 16:40:22 +0100
Subject: [PATCH] src, document-portal: add AssumedAppArmorLabel key to D-Bus
 service files

This allows sandboxed processes to activate the portal services if the
sandbox makes use of AppArmor D-Bus mediation rules that depend on the
service's security label.
---
 document-portal/org.freedesktop.portal.Documents.service.in | 1 +
 src/org.freedesktop.portal.Desktop.service.in               | 1 +
 2 files changed, 2 insertions(+)

diff --git a/document-portal/org.freedesktop.portal.Documents.service.in b/document-portal/org.freedesktop.portal.Documents.service.in
index cf0c1ef..4d0881b 100644
--- a/document-portal/org.freedesktop.portal.Documents.service.in
+++ b/document-portal/org.freedesktop.portal.Documents.service.in
@@ -2,3 +2,4 @@
 Name=org.freedesktop.portal.Documents
 Exec=@libexecdir@/xdg-document-portal
 SystemdService=xdg-document-portal.service
+AssumedAppArmorLabel=unconfined
diff --git a/src/org.freedesktop.portal.Desktop.service.in b/src/org.freedesktop.portal.Desktop.service.in
index e017386..2b31f32 100644
--- a/src/org.freedesktop.portal.Desktop.service.in
+++ b/src/org.freedesktop.portal.Desktop.service.in
@@ -2,3 +2,4 @@
 Name=org.freedesktop.portal.Desktop
 Exec=@libexecdir@/xdg-desktop-portal
 SystemdService=xdg-desktop-portal.service
+AssumedAppArmorLabel=unconfined
